src/utils/zcl_abapgit_convert.clas.locals_imp.abap

```diff
@@ -115,3 +115,678 @@ CLASS lcl_out IMPLEMENTATION.
     ENDTRY.
   ENDMETHOD.
 ENDCLASS.
+
+CLASS lcl_bcp47_language_table DEFINITION CREATE PRIVATE.
+
+  PUBLIC SECTION.
+    TYPES: BEGIN OF ty_language_mapping,
+             sap1_code  TYPE sy-langu,
+             bcp47_code TYPE string,
+           END OF ty_language_mapping,
+           tt_language_mappings TYPE STANDARD TABLE OF ty_language_mapping WITH DEFAULT KEY.
+    CLASS-DATA language_mappings TYPE tt_language_mappings.
+    CLASS-METHODS:
+      sap1_to_bcp47
+        IMPORTING
+          sap1         TYPE sy-langu
+        RETURNING
+          VALUE(bcp47) TYPE string
+        EXCEPTIONS
+          no_assignment,
+      bcp47_to_sap1
+        IMPORTING
+          bcp47       TYPE string
+        RETURNING
+          VALUE(sap1) TYPE sy-langu
+        EXCEPTIONS
+          no_assignment.
+  PROTECTED SECTION.
+  PRIVATE SECTION.
+    CLASS-METHODS fill_language_mappings.
+
+ENDCLASS.
+
+CLASS lcl_bcp47_language_table IMPLEMENTATION.
+
+  METHOD sap1_to_bcp47.
+    DATA language_mapping TYPE ty_language_mapping.
+
+    IF language_mappings IS INITIAL OR lines( language_mappings ) = 0.
+      fill_language_mappings( ).
+    ENDIF.
+
+    LOOP AT language_mappings INTO language_mapping WHERE sap1_code = sap1.
+      IF bcp47 IS INITIAL OR strlen( bcp47 ) > strlen( language_mapping-bcp47_code ).
+        bcp47 = language_mapping-bcp47_code.
+      ENDIF.
+    ENDLOOP.
+
+    IF bcp47 IS INITIAL.
+      RAISE no_assignment.
+    ENDIF.
+  ENDMETHOD.
+
+  METHOD bcp47_to_sap1.
+    DATA language_mapping TYPE ty_language_mapping.
+
+    IF language_mappings IS INITIAL OR lines( language_mappings ) = 0.
+      fill_language_mappings( ).
+    ENDIF.
+
+    LOOP AT language_mappings INTO language_mapping WHERE bcp47_code = bcp47.
+      IF sap1 IS INITIAL.
+        sap1 = language_mapping-sap1_code.
+      ENDIF.
+    ENDLOOP.
+
+    IF sap1 IS INITIAL.
+      RAISE no_assignment.
+    ENDIF.
+  ENDMETHOD.
+
+  METHOD fill_language_mappings.
+    DATA line TYPE ty_language_mapping.
+
+    line-bcp47_code = 'af'.
+    line-sap1_code = 'a'.
+    APPEND line TO language_mappings.
+    CLEAR line.
+
+    line-bcp47_code = 'sq'.
+    line-sap1_code = '뽑'.
```

Comment:
@larshp Do you have an idea here? We need these characters as they are used as SAP1 Language code. However, abaplint complains that they are non 7 bit ascii characters. Can we skip this test here somehow?

Reply:
@wurzka it's possible, but would be better to represent it in UTF. 
`line-sap1_code = cl_abap_conv_in_ce=>uccp( 'BF51' ).` 
(found by pasting your character into https://www.coderstool.com/unicode-text-converter)

Reply:
Great. Thanks @pokrakam. Will be fixed with the next commit